preshed/bloom.pyx

```diff
@@ -0,0 +1,79 @@
+# cython: infer_types=True
+# cython: cdivision=True
+#
+cimport cython
+
+from murmurhash.mrmr cimport hash64
+import math
+
+def optimal_params(members, error_rate):
+    """Calculate the optimal size in bits and number of hash functions for a
+    given number of members and error rate.  
+    """
+    base = math.log(1 / (2 ** math.log2))
+    bit_count = math.ceil((members * math.log(error_rate)) / base)
+    hash_count = math.floor((bit_count / members) * math.log(2))
+    return dict(size=bit_count, hash_funcs=hash_count)
+
+cdef class BloomFilter:
+    """Bloom filter that allows for basic membership tests.
+    
+    Only integers are supported as keys.
+    """
```

Comment:
Perhaps a class method,

```python

@classmethod
def from_error_rate(cls, members, error_rate):
    size, n_hashes = calculate_size_and_hash_count(members, error_rate)
    return cls(size, n_hashes)
```

Reply:
Being able to call `BloomFilter(len(words))` and just have it work looks pretty great. That said, I avoided putting the optimal parameter calculation in the constructor because I felt like the choice of size and hash count have very transparent effects on memory usage and speed,, while the member count and error rate do not. For example I don't really have any intuition for what the change in optimal size/hash count is if I change error rate from 1E-3 to 1E-4. 

Also, if for some reason you want to use a specific amount of memory with each filter but the constructor does optimal parameter calculation, however you end up working around that would be weird. I guess that's part of the inside-outness you mention.

I think the `from_error` class method makes sense. If a normal class instantiation interface is preferable, maybe it would make sense to have a subclass `OptimizedBloomFilter`?

Reply:
Yeah on reflection the idea of switching the constructors is trying to be much too clever. I think it's fine to just have a second constructor on a class method, called `from_error_rate`